go-packages/sim2/car.go

```diff
@@ -1,163 +1,160 @@
 package sim2
 
 import (
-  //"fmt"
-  "math"
+  "log"
+  "fmt"
 )
 
 // car - Describes routine hooks and logic for Cars within a World simulation
 
+// The distance the car should move every drive call
+const MovementPerDrive = 1.0
+
 // Car - struct for all info needed to manage a Car within a World simulation.
 type Car struct {
   // TODO determine if Car needs any additional/public members
+  id           uint
+  path         Path
+  world        CarWorldInterface
+  syncChan     chan bool
+  sendChan     *chan CarInfo
+  ethApi       EthApiInterface
+  requestState RequestState
+}
+
+type Path struct {
+  pickUp       Location
+  dropOff      Location
+  currentPos   Coords
+  currentDir   Coords
+  currentEdge  Edge
+  currentState PathState
+  routeEdges   []Edge
+  riderAddress string
+}
 
-  id uint
-  dstNode uint
-  path []uint
-  world *World
-  syncChan chan bool
-  sendChan *chan CarInfo
+type Location struct {
+  intersect Coords
+  edge Edge
 }
 
+type PathState int
+const (
+  DrivingAtRandom PathState = 0
+  ToPickUp        PathState = 1
+  ToDropOff       PathState = 2
+)
+
+type RequestState int
+const (
+  None    RequestState = 0
+  Trying  RequestState = 1
+  Success RequestState = 2
+  Fail    RequestState = 3
+)
+
 // NewCar - Construct a new valid Car object
-func NewCar(id uint, w *World, sync chan bool, send *chan CarInfo) *Car {
+func NewCar(id uint, w CarWorldInterface, ethApi EthApiInterface, sync chan bool, send *chan CarInfo) *Car {
   c := new(Car)
   c.id = id
   c.world = w
   c.syncChan = sync
   c.sendChan = send
+  c.requestState = None
+
+  c.path.currentState = DrivingAtRandom
+  newEdge := c.world.getRandomEdge()
```

Comment:
I think it would be a good idea to tie starting edge to the car's ID, since it's guaranteed to be unique. Applying randomness here means that we could potentially have cars start on the same vertex, even if it's unlikely. Determinism is good for testing :)